### PR TITLE
Feature/token timeouts

### PR DESCRIPTION
**Pull request by iStrike7**

### CODE REVIEW

File: pallets/token-timeout/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+
+// #[cfg(test)]
+// mod mock;
+// #[cfg(test)]
+// mod tests;
+
+// mod benchmarking;
+
+// pub mod weights;
+// pub use weights::WeightInfo;
```
Comment (mateuszaaa) [inline pallets/token-timeout/src/lib.rs:35]: remove

File: pallets/token-timeout/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+}
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+
+	use super::*;
+
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
```
Comment (mateuszaaa) [inline pallets/token-timeout/src/lib.rs:57]: shouldnt we start with 0?

File: pallets/token-timeout/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+
+			TimeoutMetadata::<T>::put(timeout_metadata);
+
+			Pallet::<T>::deposit_event(Event::TimeoutMetadataUpdated);
+
+			Ok(().into())
+		}
+
+		#[transactional]
+		#[pallet::weight(1_000_000_000)]
```
Comment (mateuszaaa) [inline pallets/token-timeout/src/lib.rs:184]: what about benchmarks ? will they come in separate pr ?

**Review by mateuszaaa — APPROVED** [review]
lgtm, i would encorage to add few unit test for at leat happy path scenarios

File: pallets/token-timeout/src/lib.rs (hunk 2 above)
Comment (iStrike7) [reply pallets/token-timeout/src/lib.rs]: yup

File: pallets/token-timeout/src/lib.rs (hunk 3 above)
Comment (iStrike7) [reply pallets/token-timeout/src/lib.rs]: they're here

File: pallets/token-timeout/src/lib.rs (hunk 1 above)
Comment (iStrike7) [reply pallets/token-timeout/src/lib.rs]: now used

**Pull request closed without merging**